### PR TITLE
Change asset precompilation to use development Env for plugins

### PR DESCRIPTION
**Pull request by ehelms**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.15
* [ ] 1.14
* [ ] 1.13

See Foreman's [plugin maintainer documentation](http://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

**Comment by ehelms** [comment]
This step fails when using SCSS files (http://koji.katello.org/kojifiles/work/tasks/8233/18233/build.log) but does not if run with development RAILS_ENV. Given the plugin asset task configures assets without regard for environment, we can safely set this when ew run this step.

**Comment by ehelms** [comment]
@dLobatog can you take a look at this for me? From my local testing, this should unblock Katello.

File: foreman/foreman.spec [hunk 1]
```diff
@@ -888,7 +888,7 @@ unlink tmp \\
 rm \`pwd\`/config/initializers/encryption_key.rb \\
 /usr/bin/%%{?scl:%%{scl}-}rake security:generate_encryption_key \\
 export BUNDLER_EXT_NOSTRICT=1 \\
-%%{?-s:/usr/bin/%%{?scl:%%{scl}-}rake %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=production --trace} \\
+%%{?-s:/usr/bin/%%{?scl:%%{scl}-}rake %%{-r*}%%{!?-r:plugin:assets:precompile[%%{-n*}%%{!?-n:%%{gem_name}}]} RAILS_ENV=development --trace} \\
```
Comment (ohadlevy) [inline foreman/foreman.spec]: why is this required? (/me just wondering).

Comment (ehelms) [inline foreman/foreman.spec]: See my comment above, this fixes plugin asset compilation after the sprockets-rails updates.

Comment (mmoll) [inline foreman/foreman.spec]: I have no ideas of RPMs, but with DEBs it still seems to work for plugins with RAILS_ENV=production

Comment (ehelms) [inline foreman/foreman.spec]: THis doesn't fail for all plugins, just, as far as I can tell, those that have SCSS files.

**Comment by dLobatog** [comment]
https://github.com/Katello/katello/pull/6901 
and 
https://github.com/theforeman/foreman/pull/4734

should allow Katello assets to compile again.
